grails-app/services/org/pih/warehouse/data/ProductSupplierService.groovy

```diff
@@ -355,4 +357,36 @@ class ProductSupplierService {
     void delete(String productSupplierId) {
         productSupplierGormService.delete(productSupplierId)
     }
+
+    private static void validateProductSupplierDetails(ProductSupplierDetailsCommand command) {
+        if (command.hasErrors()) {
+            throw new ValidationException("Invalid product source", command.errors)
+        }
+    }
+
+    ProductSupplier saveDetails(ProductSupplierDetailsCommand command) {
+        validateProductSupplierDetails(command)
```

Comment:
```
ProductSupplier productSupplier = new ProductSupplier(command.properties)
if (command.hasErrors()) { 
    productSupplier.errors = command.errors
    return productSupplier
}

// do some stuff

if (!productSupplier.hasErrors()) { 
    productSupplier = productSupplierGormService.save(productSupplier)
}
return productSupplier
```

Reply:
* should service throw exception 
* static vs compilestatic
    * @GrailsCompileStatic
    * @CompileStatic

Reply:
@jmiranda To add something to here - I feel like we should focus on building a proper rest API, and stick to one approach or convention (in this case I think we should stick to throwing exception or error and handling it in error controller instead returning products supplier object with errors copied from command), and not mix into it different things to "satisfy" other potential use cases (calls from other services or non API controllers)

cc @kchelstowski, @drodzewicz @alannadolny